src/nvim/api/autocmd.c

```diff
@@ -256,39 +281,73 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
   return autocmd_list;
 }
 
-/// Define an autocmd.
-/// @param opts Dictionary
-///          Required keys:
-///              event: string | ArrayOf(string)
-/// event = "pat1,pat2,pat3",
-/// event = "pat1"
-/// event = {"pat1"}
-/// event = {"pat1", "pat2", "pat3"}
+/// Define a new autocommand |autocmd|.
+///
+/// The API allows for two types of actions to be executed (mutually exclusive) when the autocommand
+/// triggers: a callback function (Lua or Vimscript), or a command (like regular autocommands).
 ///
+/// Example using callback:
+/// <pre>
+///     -- Lua function
+///     local myluafun = function() print("This buffer enters") end
+///
+///     -- Vimscript function name (as a string)
+///     local myvimfun = "g:MyVimFunction"
+///
+///     vim.api.nvim_create_autocmd({
+///       event = {"BufEnter", "BufWinEnter"},
+///       pattern = {"*.c", "*.h"},
+///       callback = myluafun,  -- Or myvimfun
+///     })
+/// </pre>
 ///
-/// -- @param {string} name - augroup name
-/// -- @param {string | table} event - event or events to match against
-/// -- @param {string | table} pattern - pattern or patterns to match against
-/// -- @param {string | function} callback - function or string to execute on autocmd
-/// -- @param {string} command - optional, vimscript command
-///          Eg. command = "let g:value_set = v:true"
-/// -- @param {boolean} once - optional, defaults to false
+/// Example using command:
+/// <pre>
+///     vim.api.nvim_create_autocmd({
+///       event = {"BufEnter", "BufWinEnter"},
+///       pattern = {"*.c", "*.h"},
+///       command = "echo 'Entering a C or C++ file'",
+///     })
+/// </pre>
 ///
-/// -- pattern = comma delimited list of patterns | pattern | { pattern, ... }
+/// Example values for patterns:
+/// <pre>
+///   pattern = "*.py,*.pyi"
+///   pattern = "*.py"
+///   pattern = {"*.py"}
+///   pattern = { "*.py", "*.pyi" }
+///   -- NOT supported
+///   pattern = {"*.py,*.pyi"}
+/// </pre>
 ///
-/// pattern = "*.py,*.pyi"
-/// pattern = "*.py"
-/// pattern = {"*.py"}
-/// pattern = { "*.py", "*.pyi" }
+/// Examples values for events:
+/// <pre>
+///   event = "FileType,CursorHold"
+///   event = "BufPreWrite"
+///   event = {"BufPostWrite"}
+///   event = {"CursorHold", "BufPreWrite", "BufPostWrite"}
+/// </pre>
 ///
-/// -- not supported
-/// pattern = {"*.py,*.pyi"}
+/// @param opts Dictionary of autocommand options:
+///             - group: (string) optional, the group name |autocmd-groups|.
```

Comment:
```suggestion
///             - group (string) optional: the group name |autocmd-groups|.
```
and similar for the other options here and below